modules/raft/src/main/java/org/apache/ignite/internal/raft/storage/segstore/IndexFileMetaArray.java

```diff
@@ -73,7 +73,15 @@ int size() {
     }
 
     long firstLogIndexInclusive() {
-        return array[0].firstLogIndexInclusive();
+        for (int i = 0; i < size; i++) {
```

Comment:
I believe that here we need the same comment. Maybe the javadoc of this method should mention what exactly it returns in various cases